x/tss/keeper/keeper.go

```diff
@@ -225,6 +225,56 @@ func (k Keeper) GetMaxSimultaneousSignShares(ctx sdk.Context) int64 {
 	return shares
 }
 
+// returns the signed blocks window to be considered when calculating the missed blocks percentage
+func (k Keeper) getSignedBlocksWindow(ctx sdk.Context) int64 {
+	var window int64
+	k.params.Get(ctx, types.KeySignedBlocksWindow, &window)
+
+	return window
+}
+
+// HasMissedTooManyBlocks returns true if the given validator address missed too many blocks within
+// the block window specifiec by this module
+func (k Keeper) HasMissedTooManyBlocks(ctx sdk.Context, address sdk.ConsAddress) bool {
+	missedBlocks, ok := k.getMissedBlocksPercent(ctx, address)
+	if !ok {
+		return false
```

Comment:
My rationalize for doing this is, if some address is not a validator, then it cannot have missed any blocks, since it is not participating in consensus.

Reply:
It's more like they missed all the blocks because missing blocks means not signing them :p After thinking a bit more about it, we recently added a check to prevent processing heartbeats from non validators. So I would treat this as an error and return `true`. Also, if the logic of `getMissedBlocksPercent` changes in the future to signal another error, then this could break.

Reply:
added a second return value of type `error` to explicitly inform that the given address is not for a validator